src/seedu/addressbook/data/CommandHistory.java

```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+
+/**
+ * Class to save the history of commands executed.
+ */
+public class CommandHistory {
+    private ArrayList<String> commandHistory = new ArrayList<>();
```

Comment:
Is there a need for an indexed list? Is there a more suitable and lax data structure that can achieve what you are doing?